tests/v2/actions/ingresses/ingress.go

```diff
@@ -0,0 +1,123 @@
+package ingresses
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	ingressActiveTimeout = 10 * time.Minute
+	ingressPollInterval  = 2 * time.Second
+	IngressTestImage     = "nginx:latest"
+	IngressTestPort      = 80
+	IngressIPDomainV25   = "sslip.io"
+	IngressIPDomainV26   = "nip.io"
+)
+
+// UpdateIngress updates an existing ingress with new specifications
+func UpdateIngress(steveClient *v1.Client, ingress *v1.SteveAPIObject, updatedIngressSpec *networkingv1.Ingress) (*v1.SteveAPIObject, error) {
```

Comment:
I believe this function would be more suitable as a shepherd extension rather than an action. Could you please check with Ugur to confirm?